src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -549,7 +549,14 @@ private BoundStatement BindLocalDeclarationStatement(LocalDeclarationStatementSy
             var binder = GetBinder(node);
```

Comment:
This is the place that ensures the right binder is used (related to my comment in PossiblyImplicitlyTypedDeconstructionLocalSymbol class).